monkey/infection_monkey/system_info/collectors/scoutsuite_collector/scoutsuite_collector.py

```diff
@@ -11,7 +11,7 @@
 def scan_cloud_security(cloud_type: PROVIDERS):
     try:
         results = run_scoutsuite(cloud_type.value)
-        if 'error' in results and results['error']:
+        if isinstance(results, dict) and 'error' in results and results['error']:
```

Comment:
When will this not be a dict? Is that case an error?

Reply:
When this is not dict, it's probably `ScoutSuiteResult` class or something similar.

Reply:
What is `ScoutSuiteResult` ? I couldn't find a class of that name.
Anyways, what I suggest is you perform whatever checks you need here to verify the type of `results`, and add a return type hint to this function. Add a matching type hint for `results` in `send_results()`

Reply:
not fixed

Reply:
I'm a little late to the game, but functions should only ever return one type.

The python way is to raise exceptions rather than check for errors like this.

Did we write `run_scoutsuite()`, or is a piece of code we don't control?

Reply:
We wrote `run_scoutsuite()` but all it does is call a function from `scoutsuite_api` which isn't part of the monkey repository.
Since we're using a fork of Scoutsuite, we can decide to create another function with a well-structured return type, but I'm not sure it's a good idea